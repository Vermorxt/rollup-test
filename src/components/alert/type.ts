import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
