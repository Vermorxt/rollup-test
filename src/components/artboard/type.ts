import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
