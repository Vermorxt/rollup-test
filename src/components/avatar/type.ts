import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
