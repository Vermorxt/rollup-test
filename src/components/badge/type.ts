import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
