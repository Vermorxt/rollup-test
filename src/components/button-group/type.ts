import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
