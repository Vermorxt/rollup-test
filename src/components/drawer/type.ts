import { HTMLProps } from 'react'
import { AnyType } from '../../_types/AnyType'
export interface Ui_DrawerProps extends HTMLProps<HTMLDivElement> {
  menuItems?: AnyType[]
  defaultOpenOnLargeScreen?: boolean
  opensRight?: boolean
  isNavbar?: boolean
  id?: string
}
