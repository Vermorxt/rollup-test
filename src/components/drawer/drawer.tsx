import { FC, forwardRef } from 'react'
import { Ui_DrawerProps } from './type'
import React from 'react'

export const Ui_Drawer: any = forwardRef<
  HTMLDivElement,
  Ui_DrawerProps & { Content: FC<Ui_DrawerProps>; Side: FC<Ui_DrawerProps> }
>(({ children, className, menuItems, defaultOpenOnLargeScreen, opensRight, id, style, ...rest }, ref) => {
  return (
    <div
      className={`
    drawer ${className ? className : ''} 
    ${defaultOpenOnLargeScreen ? 'drawer-mobile' : ''}
    ${opensRight ? 'drawer-end' : ''}
    `}
      ref={ref}
      style={style}
    >
      <input id={id} type="checkbox" className="drawer-toggle" />
      {children}
    </div>
  )
})

const Content: FC = forwardRef<HTMLDivElement, Ui_DrawerProps>(
  ({ children, className, id, isNavbar, ...rest }, ref) => {
    return (
      <div
        className={`drawer-content flex flex-col items-center ${className ? className : ''}`}
        id="main-content"
        ref={ref}
      >
        {children}
      </div>
    )
  }
)

const Side: FC = forwardRef<HTMLDivElement, Ui_DrawerProps>(({ children, className, id, ...rest }, ref) => {
  return (
    <div className={`drawer-side ${className ? className : ''}`} ref={ref}>
      <label htmlFor={id} className="drawer-overlay z-10"></label>
      {children}
    </div>
  )
})

Ui_Drawer.Content = Content
Ui_Drawer.Side = Side

export default Ui_Drawer
