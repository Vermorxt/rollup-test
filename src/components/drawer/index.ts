export { default } from './drawer'
export type { Ui_DrawerProps } from './type'
