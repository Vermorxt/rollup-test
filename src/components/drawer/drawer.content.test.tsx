import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { checkAccessibility, itSupportsSystemProps } from '../../_tests-utils'
import { Ui_DrawerProps } from './type'
import { Ui_Drawer } from './drawer'
import React from 'react'

const defaultProps: Ui_DrawerProps = {}

const selector = 'drawer-content'
describe('@cms/core/Ui_Drawer.Content', () => {
  checkAccessibility([
    <Ui_Drawer.Content {...defaultProps} ref={null}>
      Test
    </Ui_Drawer.Content>,
  ])
  itSupportsSystemProps({
    component: Ui_Drawer,
    props: defaultProps,
    refType: HTMLDivElement,
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Ui_Drawer.Content {...defaultProps} ref={null}>
        Test
      </Ui_Drawer.Content>
    )
    expect(container).toMatchSnapshot()
  })
  // NOTE: (possible debug): debugComponent(<Ui_Card.Title {...defaultProps} ref={null} />)

  it('renders default class', () => {
    const { container: available } = render(<Ui_Drawer.Content {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.${selector}`)).toHaveLength(1)
  })
})
