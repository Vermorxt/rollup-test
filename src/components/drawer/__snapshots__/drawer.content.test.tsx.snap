// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Drawer.Content should match snapshot 1`] = `
<div>
  <div
    class="drawer-content flex flex-col items-center "
    id="main-content"
  >
    Test
  </div>
</div>
`;
